lib/chef/mixin/windows_command_line_helper.rb

```diff
@@ -0,0 +1,37 @@
+#
+# Author:: Kapil Chouhan (<kapil.chouhan@msystechnologies.com>)
+# Copyright:: Copyright 2013-2020, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../win32/api/command_line_helper" if ChefUtils.windows?
+
+class Chef
+  module Mixin
+    module WindowsCommandLineHelper
+
+      if ChefUtils.windows?
+        include Chef::ReservedNames::Win32::API::CommandLineHelper
+      end
+
+      # Splits a string into an array
+      def parse_options(options)
+        return Array(options) unless ChefUtils.windows?
+
+        options.is_a?(String) ? command_line_to_argv_w_helper(options) : options
+      end
```

Comment:
This probably isn't so useful.  You don't need the check inside of parse_options that we're on windows since that'll never get called on not-windows.  And it should just be inlined in the provider, we don't really need to be extracting an abstraction around that right now.

Reply:
@lamont-granquist yes its never get called on not-windows, but it gets fail on buildkite https://buildkite.com/chef-oss/chef-chef-master-verify/builds/3460#803c3344-be0a-4e68-bd36-cf4d34beb500

Reply:
Yeah that spec either needs to be a windows-only spec or this needs to be stubbed on not-windows to turn it into a unit test.